app/drivers/adc.h

```diff
@@ -5,5 +5,11 @@
 
 #define EADC_HAL_INIT			(ADC_BASE + 0)
 #define EADC_HAL_CONFIG_CHANNEL	(ADC_BASE + 1)
+#define EADC_NO_INIT        	(ADC_BASE + 2)
+
+typedef void (*adc_conversion_ready)(const uint16_t *p_adc_values);
```

Comment:
Shouldn't the callback also say how many values are available.

Reply:
Sure.

I'm not really sure how generic adc.c should be. Right now it's super hard coded to support our use case. We might want to make the channels configurable in case we end up using another pinmux on other boards.